cmake/modules/AddSwiftStdlibOptions.cmake

```diff
@@ -0,0 +1,144 @@
+# value.  The following values are valid:
+#   - MultiThreaded (/MT)
+#   - MultiThreadedDebug (/MTd)
+#   - MultiThreadedDLL (/MD)
+#   - MultiThreadedDebugDLL (/MDd)
+if(CMAKE_BUILD_TYPE STREQUAL Debug)
+  set(SWIFT_STDLIB_MSVC_RUNTIME_LIBRARY_default MultiThreadedDebugDLL)
+else()
+  set(SWIFT_STDLIB_MSVC_RUNTIME_LIBRARY_default MultiThreadedDLL)
+endif()
+set(SWIFT_STDLIB_MSVC_RUNTIME_LIBRARY
+  ${SWIFT_STDLIB_MSVC_RUNTIME_LIBRARY_default}
+  CACHE STRING "MSVC Runtime Library for the standard library")
+
+is_build_type_optimized("${SWIFT_STDLIB_BUILD_TYPE}" swift_optimized)
+if(swift_optimized)
+  set(SWIFT_STDLIB_ASSERTIONS_default FALSE)
+else()
+  set(SWIFT_STDLIB_ASSERTIONS_default TRUE)
+endif()
+option(SWIFT_STDLIB_ASSERTIONS
+    "Enable internal checks for the Swift standard library (useful for debugging the library itself, does not affect checks required for safety)"
+    "${SWIFT_STDLIB_ASSERTIONS_default}")
+
+option(SWIFT_BUILD_RUNTIME_WITH_HOST_COMPILER
+       "Use the host compiler and not the internal clang to build the swift runtime"
+       FALSE)
+
+set(SWIFT_SDKS "" CACHE STRING
+    "If non-empty, limits building target binaries only to specified SDKs (despite other SDKs being available)")
+
+set(SWIFT_PRIMARY_VARIANT_SDK "" CACHE STRING
+    "Primary SDK for target binaries")
+set(SWIFT_PRIMARY_VARIANT_ARCH "" CACHE STRING
+    "Primary arch for target binaries")
+
+set(SWIFT_NATIVE_LLVM_TOOLS_PATH "" CACHE STRING
+    "Path to the directory that contains LLVM tools that are executable on the build machine")
+
+set(SWIFT_NATIVE_CLANG_TOOLS_PATH "" CACHE STRING
+    "Path to the directory that contains Clang tools that are executable on the build machine")
+
+set(SWIFT_NATIVE_SWIFT_TOOLS_PATH "" CACHE STRING
+   "Path to the directory that contains Swift tools that are executable on the build machine")
+
+option(SWIFT_ENABLE_MODULE_INTERFACES
+       "Generate .swiftinterface files alongside .swiftmodule files"
+       TRUE)
+
+option(SWIFT_STDLIB_ENABLE_SIB_TARGETS
+       "Should we generate sib targets for the stdlib or not?"
+       FALSE)
+
+
+set(SWIFT_DARWIN_SUPPORTED_ARCHS "" CACHE STRING
+  "Semicolon-separated list of architectures to configure on Darwin platforms. \
+If left empty all default architectures are configured.")
+
+set(SWIFT_DARWIN_MODULE_ARCHS "" CACHE STRING
+  "Semicolon-separated list of architectures to configure Swift module-only \
+targets on Darwin platforms. These targets are in addition to the full \
+library targets.")
+
+
+#
+# User-configurable Android specific options.
+#
+
+set(SWIFT_ANDROID_API_LEVEL "" CACHE STRING
+  "Version number for the Android API")
+
+set(SWIFT_ANDROID_NDK_PATH "" CACHE STRING
+  "Path to the directory that contains the Android NDK tools that are executable on the build machine")
+set(SWIFT_ANDROID_NDK_GCC_VERSION "" CACHE STRING
+  "The GCC version to use when building for Android. Currently only 4.9 is supported.")
```

Comment:
Please keep these in the main file.  Removing these from the main file would break @buttaface's build - they are building the compiler on Android and until we fix up the compiler build to cross-compile properly, these are in use.  I've been chiseling away at `AddSwift.cmake` and if I can figure out how to remove the use of these variables, we can move them at that point.

Reply:
Actually, I don't believe these three variables above are used when I'm "building the compiler on Android," as that doesn't use the Android NDK, but I believe they are used when [I cross-compile the Swift compiler itself on linux for Android](https://forums.swift.org/t/additional-linux-distributions/36188/49), as that uses the NDK.

Feel free to move stuff around and I'll fix whatever I need later, or if Saleem knows a better place to put these, go ahead with that instead.

Reply:
I wish I knew of a better place to put these.

Im worried about them being in this file and forgetting and breaking the cross-compile for the Swift compiler for android.  Id rather have the 3 flags be duplicated.  Im less concerned about the `SWIFT_ANDROID_API_LEVEL` for the compiler because the triple can encode that.  The current home grown cross-compile still requires `SWIFT_ANDROID_NDK_PATH` and `SWIFT_ANDROID_NDK_GCC_VERSION` though.  Once we have proper CMake usage for the compiler, CMake's android support will take care of that.

When you are cross-compiling the compiler, are you doing that without the standard library and therefore do not need to worry about the variables or do you do both the compiler and runtime together and need the two variables to work still?  If they do not need the variables to work for the compiler, then moving the variables is fine, otherwise we need to find a solution that will ensure that we don't break existing usages.  The shuffling should not impact the cross-compile of the runtime.

Reply:
>When you are cross-compiling the compiler, are you doing that without the standard library and therefore do not need to worry about the variables or do you do both the compiler and runtime together and need the two variables to work still?

I build [each separately](https://github.com/termux/termux-packages/blob/3eea12e61aa352986a7bdced2acf87723443ecdd/packages/swift/build.sh#L106), with its [own invocation of the build-script](https://github.com/termux/termux-packages/blob/3eea12e61aa352986a7bdced2acf87723443ecdd/packages/swift/build.sh#L143).

Either way, I think the main concern is whether this repo's built-in Android cross-compilation support, which was added to cross-compile the stdlib and then reused to cross-compile the Swift compiler itself, needs these variables for the latter.

I'm actually not sure of that for the way I use it, as the Termux package repository doesn't use the NDK itself, but the now being deprecated "standalone toolchain" generated from the NDK, which [I had to patch this repo to use instead (note that I replace a path using `SWIFT_ANDROID_NDK_GCC_VERSION` there with one that doesn't)](https://github.com/termux/termux-packages/blob/05db63787aa8e839564e8da3fa431a5e698bc90f/packages/swift/swift-cmake.patch.beforehostbuild#L148).

In sum, I'm not sure how much these variables are used to cross-compile the Swift compiler for Android too, so don't let that usage impede this reorganization or any others. I will simply add it back later if I need any of it, shouldn't be a problem.

Reply:
> I'm not sure how much these variables are used to cross-compile the Swift compiler for Android too

Okay, in that case, lets go ahead with this approach, it is definitely far more expedient, but it does need to be careful about all the other efforts going on with Swift.

Reply:
I am not going to actually remove them in any way. I am just going to move them into a file that I can include either in the main (when building together) or in the stdlib CMakeLists.txt when building the stdlib separately. So it should be completely algebraic and not cause any issues.

Reply:
Yes, this change is not going change that, but it blurs the lines a bit because there are a set of options here which are specific to the stdlib build and some which are referenced in both.  The previous state handled that by location (which granted is not very good either).  This just makes it easier to accidentally treat an option as a stdlib option which is not a pure stdlib option is all.  I was thinking about accidents that can occur in the future.